### ISSUE TITLE
YOLOv3 loss suddenly blows up

### ISSUE THREAD

**Issue by Martin-Zeman** [pjreddie/darknet#1697]
Hello,
I'm trying to train the darkent to classify signs in old Japanese manusripts. This means I have 4645 different classes. In total I have 90/10% split between my train and test set. In total there are 4054 manuscript images. A typical manuscript page has around 70-150 signs on it. Naturally, the data set is highly skewed since some characters are way more frequent than others but that's fine. I'm okay with uncommon characters not being classified accurately.

I preprocessed the dimensions of all the pictures to have be (416*n x 416*m). Basically I wanted to have the 416x416 window fit into it perfectly and filtered out any images that would have been to squased by doing so. If this decision was a good one or even a relevant one I don't really know since I'm still new to the field.

Now, my config looks like this:
```
batch=1
subdivisions=1
width=416
height=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.0001
burn_in=1000
max_batches = 9290000
policy=steps
steps=400000,450000
scales=.1,.1
...
[convolutional]
size=1
stride=1
pad=1
filters=13950
activation=linear
[yolo]
mask = 6,7,8
anchors = 10,13,  16,30,  33,23,  30,61,  62,45,  59,119,  116,90,  156,198,  373,326
classes=4645
num=9
jitter=.3
ignore_thresh = .7
truth_thresh = 1
random=1
...
[convolutional]
size=1
stride=1
pad=1
filters=13950
activation=linear
[yolo]
mask = 0,1,2
anchors = 10,13,  16,30,  33,23,  30,61,  62,45,  59,119,  116,90,  156,198,  373,326
classes=4645
num=9
jitter=.3
ignore_thresh = .7
truth_thresh = 1
random=1
```
Basically I only adjusted the filters before the yolo layer to (#classes+5)*3 as I have read in different guides. Additionally I recused batch and subdivision to 1 to prevent my GPU from crashing. I also upped the max_batches to 9290000 for which I used a rule of thumb 4645*2000. Since after a couple of minutes of training my loss blows up and reaches nan I tried decreasing the learning rate but to no avail. 

```
982: 75768.281250, 50414.855469 avg, 0.000093 rate, 0.334328 seconds, 982 images
Loaded: 0.000063 seconds
Region 82 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.000185, .5R: -nan, .75R: -nan,  count: 0
Region 94 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.000155, .5R: -nan, .75R: -nan,  count: 0
Region 106 Avg IOU: 0.010550, Class: 0.003077, Obj: 0.000000, No Obj: 0.000000, .5R: 0.000000, .75R: 0.000000,  count: 90
983: 4510973.500000, 496470.718750 avg, 0.000093 rate, 0.335505 seconds, 983 images
Loaded: 0.000080 seconds
Region 82 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.000801, .5R: -nan, .75R: -nan,  count: 0
Region 94 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.000215, .5R: -nan, .75R: -nan,  count: 0
Region 106 Avg IOU: 0.000269, Class: 0.495480, Obj: 0.157767, No Obj: 0.035802, .5R: 0.000000, .75R: 0.000000,  count: 90
984: 8605176.000000, 1307341.250000 avg, 0.000094 rate, 0.336642 seconds, 984 images
Loaded: 0.000035 seconds
Region 82 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.002420, .5R: -nan, .75R: -nan,  count: 0
Region 94 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.006351, .5R: -nan, .75R: -nan,  count: 0
Region 106 Avg IOU: -nan, Class: 0.998667, Obj: 0.994923, No Obj: 0.964727, .5R: 0.000000, .75R: 0.000000,  count: 90
985: 3321883392.000000, 333364960.000000 avg, 0.000094 rate, 0.333776 seconds, 985 images
Loaded: 0.000051 seconds
Region 82 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.002284, .5R: -nan, .75R: -nan,  count: 0
Region 94 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.003844, .5R: -nan, .75R: -nan,  count: 0
Region 106 Avg IOU: -nan, Class: 0.417355, Obj: 1.000000, No Obj: 0.936739, .5R: 0.000000, .75R: 0.000000,  count: 90
986: 931845432672256.000000, 93184843579392.000000 avg, 0.000095 rate, 0.335687 seconds, 986 images
Loaded: 0.000049 seconds
Region 82 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.000350, .5R: -nan, .75R: -nan,  count: 0
Region 94 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.000117, .5R: -nan, .75R: -nan,  count: 0
Region 106 Avg IOU: -nan, Class: 0.588889, Obj: 0.022222, No Obj: 0.648582, .5R: 0.000000, .75R: 0.000000,  count: 90
987: 393385003301367520803259938766848.000000, 39338499362996096388622654111744.000000 avg, 0.000095 rate, 0.334403 seconds, 987 images
Loaded: 0.000067 seconds
Region 82 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: nan, .5R: -nan, .75R: -nan,  count: 0
Region 94 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: nan, .5R: -nan, .75R: -nan,  count: 0
Region 106 Avg IOU: nan, Class: nan, Obj: nan, No Obj: nan, .5R: 0.000000, .75R: 0.000000,  count: 90
988: -nan, -nan avg, 0.000095 rate, 0.331415 seconds, 988 images
```

Btw I'm using a GTX 1070 with with 8GB of memory. Any hints and insights would be appreciated. Thank you!
